akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala

```diff
@@ -102,11 +105,26 @@ object ClusterShardingSettings {
       if (name == StateStoreModePersistence.name) StateStoreModePersistence
       else if (name == StateStoreModeDData.name) StateStoreModeDData
       else
-        throw new IllegalArgumentException("Not recognized StateStoreMode, only 'ddata' is supported.")
+        throw new IllegalArgumentException(
+          s"Not recognized StateStoreMode, only '${StateStoreModePersistence.name}' and '${StateStoreModeDData.name}' are supported.")
   }
   final case object StateStoreModePersistence extends StateStoreMode { override def name = "persistence" }
   final case object StateStoreModeDData extends StateStoreMode { override def name = "ddata" }
 
+  sealed trait RememberEntitiesStoreMode { def name: String }
+  object RememberEntitiesStoreMode {
+    def byName(name: String): RememberEntitiesStoreMode =
```

Comment:
if this intended to be public it's not accessible for Java because of nested object object

Reply:
Ah indeed. We have the same problem for the existing StateStoreMode